include/SDL3/SDL_intrin.h

```diff
@@ -108,29 +108,46 @@ _m_prefetch(void *__P)
 #endif
 
 #if defined(__x86_64__) || defined(_M_X64) || defined(__i386__) || defined(_M_IX86)
-# if (defined(_MSC_VER) || defined(__AVX__) || defined(SDL_HAS_TARGET_ATTRIBS)) && !defined(SDL_DISABLE_AVX)
-#  define SDL_AVX_INTRINSICS
-#  include <immintrin.h>
+# if ((defined(__GNUC__) && (__GNUC__ >= 10)) || (defined(__clang__) && __clang_major__ >= 4) || defined(_MSC_VER)) && !defined(SDL_DISABLE_RDTSC)
```

Comment:
How do you decide gcc >= 10 requirement? My gcc-4.9 has `__rdtsc()` in `ia32intrin.h` , for e.g.

Reply:
I used compiler explorer to test older versions.
Looks like around that version, gcc removed the extra underscore. (The Intel intrinsics site shows `_rdtsc()`.

I think a `SDL_rdtsc` macro would be useful, to avoid the ifdef'ery in the test suite.

Reply:
> I used compiler explorer to test older versions. Looks like around that version, gcc removed the extra underscore. (The Intel intrinsics site shows `_rdtsc()`.

I see, OK.

> I think a `SDL_rdtsc` macro would be useful, to avoid the ifdef'ery in the test suite.

Possibly better, yes

Reply:
How do you use `__rdtsc` on gcc-4.9?
This source does not compile:
```c
#include <immintrin.h>
unsigned long long f(void) {
    return __rdtsc();
}
```
[Compiler explorer link](https://godbolt.org/#g:!((g:!((g:!((h:codeEditor,i:(filename:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,selection:(endColumn:12,endLineNumber:3,positionColumn:12,positionLineNumber:3,selectionStartColumn:12,selectionStartLineNumber:3,startColumn:12,startLineNumber:3),source:'%23include+%3Cimmintrin.h%3E%0Aunsigned+long+long+f()+%7B%0A++return+__rdtsc()%3B%0A%7D'),l:'5',n:'0',o:'C%2B%2B+source+%231',t:'0')),k:51.17203365245141,l:'4',m:99.99999999999999,n:'0',o:'',s:0,t:'0'),(g:!((g:!((h:compiler,i:(compiler:g494,deviceViewOpen:'1',filters:(b:'0',binary:'1',binaryObject:'1',commentOnly:'0',demangle:'0',directives:'0',execute:'1',intel:'0',libraryCode:'0',trim:'1'),flagsViewOpen:'1',fontScale:14,fontUsePx:'0',j:1,lang:c%2B%2B,libs:!(),options:'',selection:(endColumn:1,endLineNumber:1,positionColumn:1,positionLineNumber:1,selectionStartColumn:1,selectionStartLineNumber:1,startColumn:1,startLineNumber:1),source:1),l:'5',n:'0',o:'+x86-64+gcc+4.9.4+(Editor+%231)',t:'0')),k:50.11002674183571,l:'4',m:44.01913875598086,n:'0',o:'',s:0,t:'0'),(g:!((h:output,i:(compilerName:'x86-64+gcc+12.2',editorid:1,fontScale:14,fontUsePx:'0',j:1,wrap:'1'),l:'5',n:'0',o:'Output+of+x86-64+gcc+4.9.4+(Compiler+%231)',t:'0')),header:(),l:'4',m:55.98086124401913,n:'0',o:'',s:0,t:'0')),k:48.827966347548596,l:'3',n:'0',o:'',t:'0')),l:'2',n:'0',o:'',t:'0')),version:4)

Reply:
Looks like switching between c and c++ changes behavior too.